vortex-file/src/read/layouts/chunked.rs

```diff
@@ -76,38 +79,50 @@ impl ChunkedLayoutBuilder {
     }
 
     fn metadata_layout(&self) -> VortexResult<Option<Box<dyn LayoutReader>>> {
-        self.has_metadata()
-            .then(|| {
+        self.flatbuffer()
+            .metadata()
+            .map(|m| {
+                let set_stats = BooleanBuffer::new(Buffer::from(m.bytes()), 0, Stat::cardinality())
```

Comment:
WDYT about introducing a new type to abstract away all the bit buffer manipulation?

Reply:
https://docs.rs/enumset/latest/enumset/

Reply:
it wasn't obvious how this enumset is helpful here

Reply:
not sure this crate is quite helpful, we can actually guarantee that we can append stats but the crate requires a fixed serialized repr